dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/internal/persistence/tlvformat/TLVBlockFileReaderTest.kt

```diff
@@ -0,0 +1,188 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.core.internal.persistence.tlvformat
+
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.core.internal.persistence.file.FileReaderWriter
+import com.datadog.android.core.internal.persistence.tlvformat.TLVBlockFileReader.Companion.FAILED_TO_DESERIALIZE_ERROR
+import com.datadog.android.utils.forge.Configurator
+import com.datadog.android.utils.verifyLog
+import fr.xgouchet.elmyr.annotation.IntForgery
+import fr.xgouchet.elmyr.annotation.StringForgery
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.kotlin.anyOrNull
+import org.mockito.kotlin.argumentCaptor
+import org.mockito.kotlin.eq
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import org.mockito.quality.Strictness
+import java.io.File
+import java.nio.ByteBuffer
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(Configurator::class)
+internal class TLVBlockFileReaderTest {
+    private lateinit var testedReader: TLVBlockFileReader
+
+    @Mock
+    private lateinit var mockFile: File
+
+    @Mock
+    private lateinit var mockFileReaderWriter: FileReaderWriter
+
+    @Mock
+    private lateinit var mockInternalLogger: InternalLogger
+
+    @StringForgery(regex = "^(\\w{3})\$") // a minimal number of chars to avoid flakiness
+    private lateinit var fakeDataString: String
+
+    private lateinit var fakeVersionBytes: ByteArray
+    private lateinit var fakeDataBytes: ByteArray
+    private lateinit var fakeBufferBytes: ByteArray
+
+    @BeforeEach
+    fun setup(@IntForgery(min = 0, max = 10) fakeVersion: Int) {
```

Comment:
Should it be `Short.MAX_VALUE` here?

```suggestion
    fun setup(@IntForgery(min = 0, max = Short.MAX_VALUE) fakeVersion: Int) {
```

Reply:
I think it should be `Int.MAX_VALUE`. The `Short` values are for the TLV field identifiers